src/classes/RD2_ValidationService.cls

```diff
@@ -309,7 +297,7 @@ public with sharing class RD2_ValidationService {
         validateInstallmentPeriod(rd, errorCollection);
         validateDayOfMonth(rd, errorCollection);
         validateFixedLength(rd, errorCollection);
-        validatePaymentMethod(rd, errorCollection);
+        validatePaymentMethod(new RD2_RecurringDonation(rd), errorCollection);
```

Comment:
One more suggestion - move `new RD2_RecurringDonation(rd);` to the top of this function so that we can call `isClosed` on the same instance of it. Eventually I think it'd be ideal to have all of the `validate...` functions accept an instance of `RD2_RecurringDonation` rather than the SObject itself.

Reply:
agreed. updated. I think it would take a little more work to pass it into every method since we also would want to abstract the access of the SObject values behind query methods so we can remove dot notation referencing the SObject field values directly